extractor/src/main/java/org/schabi/newpipe/extractor/utils/Parser.java

```diff
@@ -78,6 +78,37 @@ public static String matchGroup(@Nonnull final Pattern pat,
         }
     }
 
+    public static String matchGroup1MultiplePatterns(final Pattern[] patterns, final String input)
+            throws RegexException {
+        return matchMultiplePatterns(patterns, input).group(1);
+    }
+
+    public static Matcher matchMultiplePatterns(final Pattern[] patterns, final String input)
+            throws RegexException {
+        Parser.RegexException exception = null;
+        for (final Pattern pattern : patterns) {
+            final Matcher matcher = pattern.matcher(input);
+            if (matcher.find()) {
+                return matcher;
+            } else if (exception == null) {
+                // only pass input to exception message when it is not too long
+                if (input.length() > 1024) {
+                    exception = new RegexException("Failed to find pattern \"" + pattern.pattern()
+                            + "\"");
+                } else {
```

Comment:
Since the whole javascript player code is passed in, it will always exceed 1024 characters. I remember it being like more than 10k lines long

Reply:
Yeah ok, I copied this part of the code from the `parseGroup` function so I'd keep it this way.